com.unity.render-pipelines.high-definition/Editor/VFXGraph/VFXHDRPSubTarget.cs

```diff
@@ -0,0 +1,374 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using UnityEngine;
+using UnityEditor.ShaderGraph;
+using UnityEditor.ShaderGraph.Internal;
+using UnityEditor.VFX;
+
+namespace UnityEditor.Rendering.HighDefinition.ShaderGraph
+{
+    [InitializeOnLoad]
+    static class VFXHDRPSubTarget
+    {
+        internal const string Inspector = "Rendering.HighDefinition.VFXShaderGraphGUI";
+
+        static VFXHDRPSubTarget()
+        {
+            VFXSubTarget.OnPostProcessSubShader += PostProcessSubShader;
+        }
+
+        // Configures an HDRP Subshader with a VFX context.
+        private static SubShaderDescriptor PostProcessSubShader(SubShaderDescriptor subShaderDescriptor, VFXContext context, VFXContextCompiledData data)
+        {
+            var attributesStruct = GenerateVFXAttributesStruct(context, VFXAttributeType.Current);
+            var sourceAttributesStruct = GenerateVFXAttributesStruct(context, VFXAttributeType.Source);
+
+            // Defer to VFX to generate various misc. code-gen that ShaderGraph currently can't handle.
+            // We use the AdditionalCommand descriptors for ShaderGraph generation to splice these in.
+            // ( i.e. VFX Graph Block Function declaration + calling, Property Mapping, etc. )
+            GenerateVFXAdditionalCommands(
```

Comment:
This call is responsible for assembling all of the various code-gen that VFX is responsible for (graph function evaluation, assignment, etc) and organizing them into additional command descriptors.